### PR TITLE
Improve Query performance

### PR DESCRIPTION
**Pull request by kevina**
Go-datastore needs to be updated first (see https://github.com/ipfs/go-datastore/issues/54)

Also see https://github.com/libp2p/go-libp2p-kad-dht/issues/38 for context.

### CODE REVIEW

File: datastore.go [hunk 1]
```diff
@@ -72,7 +72,46 @@ func (d *datastore) Delete(key ds.Key) (err error) {
...
+		rnge = util.BytesPrefix([]byte(q.Prefix))
+	}
+	i := d.DB.NewIterator(rnge, nil)
+	return dsq.ResultsFromIterator(q, dsq.Iterator{
+		Next: func() (dsq.Result, bool) {
+			ok := i.Next()
+			if !ok {
+				return dsq.Result{}, false
+			}
+			//k := ds.NewKey(string(i.Key())).String()
```
Comment (whyrusleeping) [inline datastore.go]: Can remove these commented out things

Comment (kevina) [inline datastore.go]: Done.

**Comment by whyrusleeping** [comment]
@kevina I merged go-datastore#54 and updated it as a dependency in this repo.

**Comment by kevina** [comment]
@whyrusleeping Rebased and manually tested.  The build are failing because they can't find gx-go.

**Comment by kevina** [comment]
@whyrusleeping looks like `goveralls` needs to be fixed before this will pass.  I am assuming it will run the normal tests as there doesn't look like there in line in `.travis.yml` that will otherwise do it.

**Pull request merged**